packages/backend/src/graphql/workHour/mutations/workHourUpdateMutationField.ts

```diff
@@ -35,15 +47,46 @@ builder.mutationField('workHourUpdate', (t) =>
         hasUserId: workHour.userId,
       }
     },
-    resolve: (query, _source, { id, data: { date, duration, taskId } }) =>
-      prisma.workHour.update({
-        ...query,
-        where: { id: id.toString() },
-        data: {
+    resolve: async (query, _source, { data, date, taskId }, context) => {
+      const checkTaskInProject = await prisma.projectMembership.findFirst({
```

Comment:
In my point of view:
1. Your statement is wrong:
    > here I check taskId in the project

    No, you are checking if the user has a project membership for the project of the given task. (And this check totally makes sense)
2. Because of `1.`, the variable name `checkTaskInProject` is misleading
3. The auth check has to be inside `authScopes` (Separation of concerns principle)
4. The check you are doing is already implemented in the `isProjectMember` check

@WINt1703, can you explain me for every point where I am wrong? Because I honestly can't understand that. 🤔

Reply:
I meant in the `isProjectMember` we checking `useId` and `projectId` but we have not check for `taskId` or i don't get it.
![image](https://user-images.githubusercontent.com/41703864/205284876-84fbf3f5-71e1-49f8-8caa-59ab2b40d045.png)